### PR TITLE
Docs: Add docs for xdb-diff

### PR DESCRIPTION
**Pull request by erindru**

### CODE REVIEW

File: docs/guides/tablediff.md [hunk 1]
```diff
@@ -153,3 +153,121 @@ SQLMESH_EXAMPLE.INCREMENTAL_MODEL ONLY sample rows:
 ```
 
 The output matches, with the exception of the column labels in the `COMMON ROWS sample data differences`. The underlying table for each column is indicated by `s__` for "source" table (first table in the command's colon operator `:`) and `t__` for "target" table (second table in the command's colon operator `:`).
+
+## Diffing tables or views across gateways
+
+!!! info "Enterprise feature"
```
Comment (sungchun12) [inline docs/guides/tablediff.md:159]:
```suggestion
!!! info "Tobiko Cloud Feature"
```

File: docs/guides/tablediff.md [hunk 2]
```diff
@@ -153,3 +153,121 @@ SQLMESH_EXAMPLE.INCREMENTAL_MODEL ONLY sample rows:
...
+
+    Cross-gateway table diffing is a feature of [SQLMesh Enterprise](./observer.md#installation).
```
Comment (sungchun12) [inline docs/guides/tablediff.md:161]:
```suggestion
    Cross-gateway table diffing is a feature of [Tobiko Cloud](./observer.md#installation).
```

File: docs/guides/tablediff.md [hunk 3]
```diff
@@ -153,3 +153,121 @@ SQLMESH_EXAMPLE.INCREMENTAL_MODEL ONLY sample rows:
 ```
 
 The output matches, with the exception of the column labels in the `COMMON ROWS sample data differences`. The underlying table for each column is indicated by `s__` for "source" table (first table in the command's colon operator `:`) and `t__` for "target" table (second table in the command's colon operator `:`).
+
+## Diffing tables or views across gateways
```
Comment (sungchun12) [inline docs/guides/tablediff.md:157]: Can you add a screenshot of a shell output where you run a command and see what exact matches and diffs should look and feel like?

Comment (sungchun12) [reply docs/guides/tablediff.md:157]: Also had generic metrics on performance time for table size. Ex:
- 1 billion rows across snowflake and bigquery, 50 seconds, etc.

Comment (sungchun12) [reply docs/guides/tablediff.md:157]: Update: no need to add metrics for now

Comment (erindru) [reply docs/guides/tablediff.md:157]: I've added some sample output as text to keep with the existing theme of this page (and also text is easier to remove my name from the schema references)

Comment (sungchun12) [reply docs/guides/tablediff.md:157]: Looks Gouda

**Review by treysp — APPROVED** [review]

**Pull request merged**